### PR TITLE
Feature/raidpackage add all consumable options

### PR DESCRIPTION
**Pull request by uino59**
Added all consumable options and some error handling. Fixes issue #9

### CODE REVIEW

**Review by msadowskigraduate — CHANGES_REQUESTED** [review]
add `*__pycache__` to .gitignore. It contains preloaded bytecode that is being created by python runtime. There is no need to commit that.

Besides that v.impressive

File: raidpackage.py [hunk 1]
```diff
@@ -137,19 +117,31 @@ async def process_user_quantity_input(client, usr, item, qtyMax, usr_id):
...
+        error_msg = await usr.send(f"You must enter a number - please unclick and reclick the {item} emoji")
+        _error_messages.append(error_msg)
     else:
         await botMsg.delete()
+        
+        if len(_error_messages) > 0:
+            for error_msg in _error_messages:
+                await error_msg.delete()
+                _error_messages.remove(error_msg)
         return int(msg.content)
```
Comment (msadowskigraduate) [inline raidpackage.py:147]: Here you are not checking if `int(msg.content) <= maxQty`

**Review by msadowskigraduate — CHANGES_REQUESTED** [review]

File: item_definitions.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        json_item_data = json.load(f)
+
+    i = 0
+    for json_object in json_item_data:
+        item = Item(**json_object)
+        items.append(item)
+        i += 1
+
+    for item in items:
+        print(item.item_name)
```
[the remark this answers is not in the document]
Comment (uino59) [reply item_definitions.py:30]: This was just for proof of concept whilst developing so I knew it was functioning and should have been removed

File: item_definitions.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        self.item_category = item_category
+        self.item_max = item_max
+        self.item_emoji = item_emoji
+
+
+def populate_items():
+    with open('resources/items.json') as f:
+        json_item_data = json.load(f)
+
+    i = 0
```
[the remark this answers is not in the document]
Comment (uino59) [reply item_definitions.py:23]: Removed

File: item_definitions.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+from discord import message
+
+items = []
+
+REACTION_ACCEPT='✅'
+REACTION_CANCEL='❌'
+REACTION_NEW_ORDER='📝'
+
+
+# Utility Pots
```
[the remark this answers is not in the document]
Comment (uino59) [reply item_definitions.py]: Removed

File: main.py [hunk 5]
```diff
@@ -15,8 +16,11 @@
 storage = {}
 initMsgId = 0
 
+populate_items()
```
[the remark this answers is not in the document]
Comment (uino59) [reply main.py:19]: Agreed, moved

File: on_guild_join.py [hunk 6]
```diff
@@ -10,14 +10,15 @@ async def intialize_order_channels(guild, client, confirmed_channel_name, order_
...
         if chnl.name == order_channel_name:
             order_channel = chnl
-            order_channel.purge(check=lambda m: m.author == client.user)
+            await order_channel.purge(check=lambda m: m.author == client.user)
             continue
 
     if confirmed_order_channel == None:
         permissions = {
-             guild.default_role: discord.PermissionOverwrite(send_messages=False)
+            guild.default_role: discord.PermissionOverwrite(send_messages=False),
```
[the remark this answers is not in the document]
Comment (uino59) [reply on_guild_join.py:18]: Removed

File: raidpackage.py [hunk 7]
```diff
@@ -35,94 +41,66 @@ async def listen(client, sent_message, preorder_embed, usr, confirmed_channel):
...
-    if str(reaction_remove_payload.emoji) == icon_definitions.emoji_spectral_int:
-        await sent_message.edit(embed=preorder_embed.set_field_at(1, name="Potion", value='None', inline=False))
+async def wait_for_order_reaction_add(payload, client, sent_message, preorder_embed, usr, order_channel):  
+    for item in item_definitions.items:
+        if str(payload.emoji) == item.item_emoji:
+            qtyReq = await process_user_quantity_input(client, usr, item.item_name, item.item_max, payload.user_id) 
+            if qtyReq == None:
+                return
+            else:
+                await sent_message.edit(embed=preorder_embed.set_field_at(item.position_id, name=item.item_category,
```
[the remark this answers is not in the document]
Comment (uino59) [reply raidpackage.py:95]: Break added

File: raidpackage.py (hunk 1 above)
Comment (uino59) [reply raidpackage.py]: Fixed in commit 1f34b26c96d2f7f135f263d8981e735e55197e61

File: item_definitions.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import json
+
+from discord import message
+
+items = []
```
[the remark this answers is not in the document]
Comment (uino59) [reply item_definitions.py:4]: Opened an issue to look into it later

File: icon_definitions.py [hunk 9]
```diff
@@ -2,9 +2,29 @@
 REACTION_CANCEL='❌'
 REACTION_NEW_ORDER='📝'
 
-emoji_shaded_weightstone='<:ShadedWeight:873183450063069194>'
-emoji_shaded_sharpen='<:ShadedSharpen:873183190880235530>'
-emoji_spectral_int='<:PotSpectralInt:873187413424484402>'
-emoji_spectral_str='<:PotSpectralStr:873183439656996895>'
-# emoji_spectral_agi='<:PotSpectralInt:873187413424484402>'
-# emoji_spectral_stam='<:PotSpectralInt:873187413424484402>'
+# Combat Pots
```
[the remark this answers is not in the document]
Comment (uino59) [reply icon_definitions.py]: Done

**Pull request merged**